### PR TITLE
fix maximum fairy souls on stranded and remove unnecessary values for misc

### PR DESCRIPTION
**Pull request by Shiiyu**
Changes maximum fairy souls on stranded to 3.

### CODE REVIEW

**Review by metalcupcake5 — APPROVED** [review]
look god no test

**Comment by dukio** [comment]
This shouldn't be in stats.ejs imo, that's just for displaying stats... 

I would implement the same logic in lib.js#L1710

```js
  output.fairy_souls = {
    collected: userProfile.fairy_souls_collected,
    total: MAX_SOULS,
    progress: Math.min(userProfile.fairy_souls_collected / MAX_SOULS, 1),
  };
```

---

Edit... hold on why the frick is this in lib.js
image

this should be in constants... !??!

**Review by nstringham — CHANGES_REQUESTED** [review]
Similar to dukio, I think we should be changing the value of `calculated.fairy_souls.total` instead of putting logic in the template.

**Review by nstringham — APPROVED** [review]

File: src/constants/misc.js [hunk 1]
```diff
@@ -422,3 +422,5 @@ export const increase_most_stats_exclude = [
   "foraging_fortune",
   "pristine",
 ];
+
+export const MAX_SOULS = 228;
```
Comment (dukio) [inline src/constants/misc.js]: This could be:

```js
export const fairy_souls: {
  max: {
    normal: 228,
    stranded: 3,
  }
}
```
Because right now the "3" for stranded is still hardcoded in lib.js

So in lib.js you can do:
```js
const totalSouls = profile.game_mode === "island"
  ? constants.fairy_souls.max.stranded
  : constants.fairy_souls.max.normal;
```

File: src/lib.js [hunk 2]
```diff
@@ -1704,11 +1703,12 @@ export const getStats = async (
...
     }
   }
+  const totalSouls = profile.game_mode === "island" ? 3 : constants.MAX_SOULS;
 
   output.fairy_souls = {
     collected: userProfile.fairy_souls_collected,
-    total: MAX_SOULS,
-    progress: Math.min(userProfile.fairy_souls_collected / MAX_SOULS, 1),
+    total: totalSouls,
+    progress: Math.min(userProfile.fairy_souls_collected / totalSouls, 1),
```
Comment (dukio) [inline src/lib.js]: I noticed we are calculating progress but not displaying it 🤔 
Maybe it should be a tooltip on the image thing

Btw idk if the `Math.min()` is still useful cause they fixed having more than max fairy souls by giving each fairy soul an uuid. So if the progress is > 1 it's our fault for having an outdated `constants.fairy_souls` max value

**Comment by Shiiyu** [comment]
If the only request change is the text format on fairy souls tippy I will not request a review again.

File: src/lib.js [hunk 3]
```diff
@@ -1704,11 +1703,14 @@ export const getStats = async (
...
+  const totalSouls = profile.game_mode === "island"
+    ? constants.MAX_SOULS.stranded
+    : constants.MAX_SOULS.normal;
 
   output.fairy_souls = {
     collected: userProfile.fairy_souls_collected,
-    total: MAX_SOULS,
-    progress: Math.min(userProfile.fairy_souls_collected / MAX_SOULS, 1),
+    total: totalSouls,
+    progress: (userProfile.fairy_souls_collected / totalSouls),
```
Comment (dukio) [inline src/lib.js]: Looks good but it's not formatted (prettier)... the parenthesis here are useless

File: src/constants/misc.js [hunk 4]
```diff
@@ -254,26 +254,11 @@ export const ranks = {
     tag: "BUILD TEAM",
   },
 
-  MODERATOR: {
-    color: "2",
-    tag: "MOD",
-  },
-
```
Comment (dukio) [inline src/constants/misc.js:261]: This is not related to this PR though...

File: src/constants/misc.js [hunk 5]
```diff
@@ -254,26 +254,11 @@ export const ranks = {
...
-  HELPER: {
-    color: "9",
-    tag: "HELPER",
-  },
-
-  "JR HELPER": {
-    color: "9",
-    tag: "JR HELPER",
-  },
-
```
Comment (dukio) [inline src/constants/misc.js:276]: This is not related to this PR though...

**Comment by dukio** [comment]
Apart from the little things I commented on looks good to me... (will approve when checks pass)

File: src/lib.js (hunk 2 above)
Comment (Shiiyu) [reply src/lib.js]: image
image

File: src/constants/misc.js (hunk 5 above)
Comment (Shiiyu) [reply src/constants/misc.js:276]: image
image

**Review by nstringham — CHANGES_REQUESTED** [review]
I don't see why `ranks` is modified in this PR and you need to fix the prettier issue but other than that this looks good
